catalog/models.py

```diff
@@ -283,6 +287,10 @@ def get_meets_with_subjects(self):
     corequisites = models.TextField(null=True)
     either_prereq_or_coreq = models.BooleanField(default=False)
 
+    # EECS subject renumbering
+    old_id = models.TextField(null=True)
```

Comment:
1. This should be a `CharField` with an appropriate `max_length`
2. Is there a reason to save the `new_id` separately? Won't the catalog always store the new ID as the subject ID?

Reply:
1. Thanks for pointing this out; I have fixed it in b0fd96bd2c.
2. This is mainly for backwards-compatibility with the spring 2022 schedule (which still has the old numbers) and with existing `.road` files.

Reply:
I think backwards-compatibility should still work without this field as long as the new ID is always stored as the subject_id, right? We would just have to make sure any old roads are upgraded to the new system when they are returned from the API, likely by checking if the subjects have the new_id field, and if not then replacing them with the new subject.

Reply:
I think so, but this would entail retroactively updating the spring 2022 catalog (on the FireRoad server) and old roads (in CourseRoad). I suppose that this is not a bad thing as I previously thought, given that CourseRoad uses the latest catalog (which will soon have the new numbers) in any case.

I'll change this to update `subject_id` with the new ID.